client/homebrew/pages/sharePage/sharePage.jsx

```diff
@@ -48,6 +55,13 @@ const SharePage = createClass({
 		}
 	},
 
+	messageReceived : function(e){
+		console.log(`Message RX: ${new Date}`);
+		if(e.origin === window.location.origin && e.data === `brewUpdate:${this.props.brew.shareId}`) {
```

Comment:
Theoretically, the same origin check shouldn't be necessary since by definition only same origin tabs (etc) can communicate. Did you find otherwise?

https://developer.mozilla.org/en-US/docs/Web/API/BroadcastChannel

> The BroadcastChannel interface represents a named channel that any [browsing context](https://developer.mozilla.org/en-US/docs/Glossary/Browsing_context) of a given [origin](https://developer.mozilla.org/en-US/docs/Glossary/Origin) can subscribe to. It allows communication between different documents (in different windows, tabs, frames or iframes) of the same origin.

Reply:
The origin check was left over from previous experimentation with [window.postMessage](https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage); after some time and a distinct lack of success, I realized that BroadcastChannel was likely the better candidate for what I was trying to achieve.

I should also remove that unnecessary `console.log` call, it was only to show that messages were being received.

Reply:
I've removed the console log, and all of the received message checks, instead using JSON.stringify and JSON.parse to pass the entire `brew` from EditPage directly to the SharePage, which is then applied as a state update, causing a re-render, rather than a reload.

As predicted, this eliminates the loss of place in the brew on the SharePage when an update comes through from EditPage.